sanic/models/futures.py

```diff
@@ -1,39 +1,46 @@
-from collections import namedtuple
-
-
-FutureRoute = namedtuple(
-    "FutureRoute",
-    [
-        "handler",
-        "uri",
-        "methods",
-        "host",
-        "strict_slashes",
-        "stream",
-        "version",
-        "name",
-        "ignore_body",
-        "websocket",
-        "subprotocols",
-        "unquote",
-        "static",
-    ],
-)
-FutureListener = namedtuple("FutureListener", ["listener", "event"])
-FutureMiddleware = namedtuple("FutureMiddleware", ["middleware", "attach_to"])
-FutureException = namedtuple("FutureException", ["handler", "exceptions"])
-FutureStatic = namedtuple(
-    "FutureStatic",
-    [
-        "uri",
-        "file_or_directory",
-        "pattern",
-        "use_modified_since",
-        "use_content_range",
-        "stream_large_files",
-        "name",
-        "host",
-        "strict_slashes",
-        "content_type",
-    ],
-)
+from pathlib import PurePath
+from typing import NamedTuple, List, Union, Callable
+
+
+class FutureRoute(NamedTuple):
+    handler: str
+    uri: str
+    methods: Union[None, List[str]]
+    host: str
+    strict_slashes: bool
+    stream: bool
+    version: Union[None, int]
+    name: str
+    ignore_body: bool
+    websocket: bool
+    subprotocols: Union[None, List[str]]
+    unquote: bool
+    static: bool
+
+
+class FutureListener(NamedTuple):
+    listener: Callable
```

Comment:
We do have defined types that are more specific than just `Callable`. I think they are currently defined in `./handlers.py`, but could probably find a nicer home.

Reply:
Relocating this into a common model under `sanic.models.handler_types` so that we can extend them as required. This comes in like with how we are managing the `protocol_types` as well.